iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/CertificateBasedAuthenticationCheck.java

```diff
@@ -19,35 +19,69 @@
+  private static final String ALLOWING_NO_CERTIFICATE_MESSAGE = "Connections without client certificates will be permitted. Make sure it is safe here.";
+
+  private static final List<String> PATH_CONTAINER_APPS = ArmTreeUtils.computePath("configuration/ingress");
 
   @Override
   protected void registerResourceConsumer() {
-    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", CertificateBasedAuthenticationCheck::checkNegotiateClientCertificate);
+    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", CertificateBasedAuthenticationCheck::checkHostnameConfigurations);
+    register("Microsoft.App/containerApps", CertificateBasedAuthenticationCheck::checkContainerApps);
   }
 
-  private static void checkNegotiateClientCertificate(CheckContext ctx, ResourceDeclaration resource) {
-    PropertyUtils.value(resource, "negotiateClientCertificate")
+  private static void checkHostnameConfigurations(CheckContext ctx, ResourceDeclaration resource) {
+    PropertyUtils.get(resource, "negotiateClientCertificate")
       .ifPresentOrElse(
-        value -> {
-          if (isBooleanFalse(value)) {
-            ctx.reportIssue(value, DISABLED_CERTIFICATE);
+        property -> {
+          if (isBooleanFalse(property.value())) {
+            ctx.reportIssue(property, DISABLED_CERTIFICATE_MESSAGE);
           }
         },
-        () -> ctx.reportIssue(resource.type(), String.format(MISSING_CERTIFICATE, "negotiateClientCertificate")));
+        () -> ctx.reportIssue(resource.type(), String.format(MISSING_CERTIFICATE_MESSAGE, "negotiateClientCertificate")));
+  }
+
+  private static void checkContainerApps(CheckContext ctx, ResourceDeclaration resource) {
+    if (isResourceVersionBefore(resource, "2022-10-01")) {
+      List<Tree> resolvedTrees = ArmTreeUtils.resolveTreePath(resource, new LinkedList<>(PATH_CONTAINER_APPS));
+      for (Tree resolvedTree : resolvedTrees) {
+        PropertyUtils.get(resolvedTree, "clientCertificateMode")
+          .ifPresentOrElse(
+            property -> {
+              if (isStringValue(property.value(), "accept")) {
+                ctx.reportIssue(property, ALLOWING_NO_CERTIFICATE_MESSAGE);
+              } else if (isStringValue(property.value(), "ignore")) {
+                ctx.reportIssue(property, DISABLED_CERTIFICATE_MESSAGE);
+              }
+            },
+            () -> ctx.reportIssue(resolvedTree, String.format(MISSING_CERTIFICATE_MESSAGE, "clientCertificateMode")));
+      }
+    }
   }
 
   private static boolean isBooleanFalse(Tree tree) {
     return ((ArmTree) tree).is(ArmTree.Kind.BOOLEAN_LITERAL) && !((BooleanLiteral) tree).value();
   }
+
+  private static boolean isStringValue(Tree tree, String value) {
+    return ((ArmTree) tree).is(ArmTree.Kind.STRING_LITERAL) && ((StringLiteral) tree).value().equals(value);
+  }
```

Comment:
```suggestion
  private static boolean isString(Tree tree, Predicate<String> predicate) {
    return TextUtils.matchesValue(tree, predicate).isTrue();
  }
```
FYI I replaced this later with a more generic method + added the usage of TextUtils which I tend to forget :sweat_smile:
I can also adapt this PR if you think it make sense.

Reply:
My naming suggestion would be `isValue()` or `hasValue()` because you are not testing if the tree is of type string but if the tree is a string with an expected value.